### ISSUE TITLE
[BUG] The `skin_url` option does not use the `suffix` option: the `.min` suffix is always used

### ISSUE THREAD

**Issue by Rezyan** [tinymce/tinymce#8131]
**What is the current behavior? Describe the bug**

When using the `skin_url` option, the `suffix` option is not used to create the URLs: the `.min` suffix is always used. This occurs because of the following lines:

https://github.com/tinymce/tinymce/blob/a48c85b693f8ce3690a853d5fbd6b6ebad7ab23a/modules/tinymce/src/themes/silver/main/ts/ui/skin/Loader.ts#L18

https://github.com/tinymce/tinymce/blob/a48c85b693f8ce3690a853d5fbd6b6ebad7ab23a/modules/tinymce/src/themes/silver/main/ts/ui/skin/Loader.ts#L25

https://github.com/tinymce/tinymce/blob/a48c85b693f8ce3690a853d5fbd6b6ebad7ab23a/modules/tinymce/src/themes/silver/main/ts/ui/skin/Loader.ts#L36

**Please provide the steps to reproduce and if possible a minimal demo of the problem via [fiddle.tiny.cloud](https://fiddle.tiny.cloud/) or similar.**

```js
tinymce.init({
	skin_url: '/custom/path/to/skin',
	// ...
});
```

**What is the expected behavior?**

The code below should load `/custom/path/to/skin/skin.css` and `/custom/path/to/skin/content.css`.
```js
tinymce.init({
	skin_url: '/custom/path/to/skin',
	// suffix: '.min',
	// ...
});
```

The code below should load `/custom/path/to/skin/skin.min.css` and `/custom/path/to/skin/content.min.css`.
```js
tinymce.init({
	skin_url: '/custom/path/to/skin',
	suffix: '.min',
	// ...
});
```

**Which versions of TinyMCE, and which browser / OS are affected by this issue? Did this work in previous versions of TinyMCE?**

- *TinyMCE version*: `6.2.0`
- *Browsers and OS*: all
- *Worked in previous versions of TinyMCE*: not as far as I know

**Possible solution**

In my opinion, there would be 2 solutions:
1. Use the `suffix` option with the `skin_url` option.
2. Create several options to override the default basenames:
    - `skin.css`, `skin.min.css`
    - `skin.shadowdom.css`, `skin.shadowdom.min.css`
    - `content.css`, `content.min.css`
    - `content.inline.css`, `content.inline.min.css`

**Comment by TheSpyder** [comment]
The documentation is fairly clear that it only applies to JavaScript files:
https://www.tiny.cloud/docs/tinymce/6/editor-important-options/#suffix

> When loading things like the theme and plugins this suffix will be inserted into all requests

However I think this is a valid feature request, and might fit within some other plans we have for the 6.4 release.

**Comment by lostkeys** [comment]
Hi @Bugophobia. I've added it to our feature request prioritization board.

Cheers,
— Fredrik, Product Manager

**Comment by EvilJordan** [comment]
Yo, so this is all super-broken in v7.1.1.

Omitting a `suffix: '.min'` property in the `init` while including `tinymce.min.js` results in all sorts of 404 errors as the parsed URLs look like this: `'https://.../tinymce/icons/default/iconsundefined.js`

Additionally, if one tries to include the un-minified `tinymce.js` without the `suffix` declaration, one also gets the 404 errors.

Attempting to add: `suffix: ''` still results in `'https://.../tinymce/icons/default/iconsundefined.js` for un-minified `tinymce.js` include.

**Comment by TheSpyder** [comment]
@EvilJordan that might've been a 7.1.0 change, does it still work in 7.0.1? The specific code that generates the `icons` url hasn't changed in 4 years, so it must be something else we changed.

The suffix is supposed to default to empty string, is this happening all the time or do you have a specific replication case you can share?

**Comment by EvilJordan** [comment]
This is my first attempt ever at using TinyMCE, so idk about previous versions.

It's happening all the time on a vanilla node + express implementation.

To be clear, it's not just `icons`, it's everything: `modules`, `theme`, etc. Anything that gets pulled in with the `suffix` variable.

**Comment by spocke** [comment]
@EvilJordan What is you exact `tinymce.init` config are you using the `skin_url` option mentioned in the original ticket?

**Comment by EvilJordan** [comment]
I am not using `skin_url`. This is the config that _does not work_:
```
tinymce.init({
	base_url: '/tinymce',
	license_key: 'gpl',
	selector: 'textarea#tinymce',
	plugins: 'save lists link table code help wordcount',
	toolbar: 'save cancel | undo redo | blocks | bold italic | alignleft aligncenter alignright alignjustify | indent outdent'
});
```

This is the config that _does_ work:
```
tinymce.init({
	base_url: '/tinymce',
	license_key: 'gpl',
	selector: 'textarea#tinymce',
	suffix: '.min',
	plugins: 'save lists link table code help wordcount',
	toolbar: 'save cancel | undo redo | blocks | bold italic | alignleft aligncenter alignright alignjustify | indent outdent',
});
```

**Comment by TheSpyder** [comment]
@EvilJordan try removing `base_url`. The editor should be able to pick that up automatically from the script tag used to load it.

**Comment by EvilJordan** [comment]
> @EvilJordan try removing `base_url`. The editor should be able to pick that up automatically from the script tag used to load it.

Same result (`undefined` in URLs). I tried removing both the `base_url` and the `suffix` properties, together and individually.

**Comment by kimwoodfield** [comment]
Marking this as closed since a solution to the original problem outlined in the description was implemented in [our 7.1 release](https://www.tiny.cloud/docs/tinymce/latest/7.1-release-notes/#overview).
